www/src/context/SettingsContext.jsx

```diff
@@ -1,7 +1,9 @@
 import React, { createContext, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { Helmet } from 'react-helmet';
+import * as Components from '~paragon-react'; // eslint-disable-line
 import { THEMES } from '../../theme-config';
+import getParagonComponentsTypes from '../utils/getParagonComponentsTypes';
```

Comment:
Hmm, these feel a bit strange to be adding to a `SettingsContext` which should be bounded to the context of what settings are available for the user in the "Settings" `Sheet` by clicking the gear icon in the header.

Also, `SettingsContext` wraps the entire Gatsby application, where presumably the `getParagonComponentsTypes` is only relevant to the "Insights" page. Should there be an `InsightsContext` specific to the "Insights" page if it needs to be placed in a context instead?

Reply:
Yes, correct, I added `InsightsContext`. Is it good idea to wrap also whole application? I just think of `getParagonComponentsTypes` is loaded before user goes to `/insights` so that page will be rendered faster. Is it correct?